### ISSUE TITLE
Multiple entities forming one entity

### ISSUE THREAD

**Issue by EtienneBruines** [paked/engi#95]
Imagine one having an `Entity`, with a `RenderComponent` - let's say this is a character in-game. 
Now, let's say that this character has a health-bar, a bar that's more/less green, right above his head. 

The health-bar is associated with the character, and therefore would be the same `Entity`. But as there's only room for one `RenderComponent` for that `Entity`, it should become another `Entity`. 

What's the best way to handle this? Two `Entity`s that are linked together? (i.e. a `ReferenceComponent`, which references to the other `Entity`), or is there a way we can say that that `Entity` has two `RenderComponent`s? What's the best-practice here? 

@paked @matiwinnetou ?

**Comment by paked** [comment]
Most other engines implement this as a group.

However, I have no idea how this looks from an ECS standpoint. Perhaps things are just bound together because they are part of the same system?

**Comment by EtienneBruines** [comment]
An `Entity` mainly being a `[]Component`, a group would roughly be `[][]Component`. 

I could imagine we add another `Update` function?

```go
func (s *Systemer) Update(group *engi.Group, dt float32) {
    var (
        rcs []*RenderComponent
        spaces []*SpaceComponent
        ok bool
    )
  
    if rcs, ok = group.ComponentsFast(rcs).([]*RenderComponent); !ok {
        return
    }
}
```

(Note that we probably cannot cast to `[]*RenderComponent`, but for the sake of argument, let's say we could)

**Comment by paked** [comment]
@EtienneBruines I think the best way to do this would be with a group of `Entity`s, as opposed to `[][]Component`.

**Comment by EtienneBruines** [comment]
According to the stuff I've read on Stackoverflow and forums of other engines, the solution they talk about, is to have the user/programmer implement some kind of `ParetnComponent`, or `HealthBarEntityComponent`, which holds a reference (pointer) to the linked `Entity`. This means that the `HealthBar` `Entity` doesn't have to be part of the `System` which manages the Health - but when the health-bar changes (when the `HealthSystem` has processed the `playerEntity`), the referenced `HealthBar` `Entity` will be changed (i.e. the `RenderComponent`). 

Long story short: the user should implement this.
